app/server/src/features/accounts/methods.unit.test.ts

```diff
@@ -0,0 +1,463 @@
+                },
+                {
+                    organization: 'School of Public Policy',
+                },
+                {
+                    organization: 'Southern California',
+                },
+            ],
+        };
+
+        prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+        const expectedOutput = [
+            'Prytaneum',
+            'University of California, Riverside',
+            'School of Public Policy',
+            'Southern California',
+        ];
+
+        const output = await AccountMethods.findOrgsByUserId(userData.id, prismaMock);
+
+        expect(output).toEqual(expectedOutput);
+    });
+
+    test('should not find orgs', async () => {
+        prismaMock.user.findUnique.mockRejectedValueOnce(null);
+
+        await expect(AccountMethods.findOrgsByUserId(userData.id, prismaMock)).rejects.toBeNull();
+    });
+});
+
+/*
+ *  test cases for the function called when a user is registering themselves
+ */
+describe('registerSelf', () => {
+    test('should register a user if passwords match', async () => {
+        const input = {
+            email: userData.email,
+            firstName: userData.firstName,
+            lastName: userData.lastName,
+            password: userTextPassword,
+            confirmPassword: userTextPassword,
+        };
+
+        // mock registered user
+        const registeredUser = { ...userData };
+
+        prismaMock.user.create.mockResolvedValueOnce(registeredUser);
+
+        const userWithGlobalId = toUserId(registeredUser);
+        const token = await jwt.sign({ id: userWithGlobalId.id });
+
+        const expectedOutput = { registeredUser, token };
+
+        const output = await AccountMethods.registerSelf(prismaMock, input);
+
+        expect(output).toEqual(expectedOutput);
+    });
+
+    test('should throw if passwords do not match', async () => {
+        const input = {
+            email: userData.email,
+            firstName: userData.firstName,
+            lastName: userData.lastName,
+            password: 'password',
+            confirmPassword: 'mismatchedPassword',
+        };
+
+        await expect(AccountMethods.registerSelf(prismaMock, input)).rejects.toThrow('Passwords must match');
+    });
+});
+
+/*
+ *  test cases for logging in a user and returning the user and a token to be used as a cookie
+ */
+describe('loginWithPassword', () => {
+    test('should throw if no password exists', async () => {
+        const user = {
+            ...userData,
+            password: null,
+        };
+
+        // mock input for a login form
+        const input = {
+            email: userData.email,
+            password: 'nonNullPassword',
+        };
+
+        prismaMock.user.findUnique.mockResolvedValueOnce(user);
+
+        await expect(AccountMethods.loginWithPassword(prismaMock, input)).rejects.toThrow(
+            'Login failed; Invalid email or password.'
+        );
```

Comment:
This one is a simpler change
```ts
// arrange
  const user = {
            ...userData,
            password: null,
        };
        // mock input for a login form
        const input = {
            email: userData.email,
            password: 'nonNullPassword',
        };
        prismaMock.user.findUnique.mockResolvedValueOnce(user);

// assert
        await expect(AccountMethods.loginWithPassword(prismaMock, input)).rejects.toThrow(
            'Login failed; Invalid email or password.'
        );
```

I think the `act` and `assert` blocks get combined here.  Not 100% sure unless we can save the thrown error somehow? Like `const error = await expect(...);` or `const error = await expect(...).catch(idk what to put here);`

Reply:
I couldn't really find a way to maintain AAA when it came to throwing errors. If you do find a way to split it up like that let me know and I can change it.

Reply:
Also just to note that best practices guide you sent seems to promote the way I'm doing it currently https://github.com/goldbergyoni/javascript-testing-best-practices#-%EF%B8%8F-110-dont-catch-errors-expect-them